### PR TITLE
Use dotnet nuget instead of nuget.exe

### PR DESCRIPTION
**Pull request by thomaslevesque**

### CODE REVIEW

File: FakeItEasy.Deploy/Program.cs [hunk 1]
```diff
@@ -109,7 +109,7 @@ static async Task UploadPackageToNuGetAsync(string path, string nugetServerUrl,
         return;
     }
 
-    await RunAsync(ToolPaths.NuGet, $"push \"{path}\" -ApiKey {nugetApiKey} -Source {nugetServerUrl} -NonInteractive -ForceEnglishOutput", noEcho: true);
+    await RunAsync("dotnet", $"nuget push \"{path}\" --api-key {nugetApiKey} --source {nugetServerUrl} --force-english-output", noEcho: true);
```
Comment (thomaslevesque) [inline FakeItEasy.Deploy/Program.cs:112]: The `-NonInteractive` flag has been reversed in `dotnet nuget`, it's non-interactive by default and there's a `--interactive` flag.

Also, there's a `--no-service-endpoint` flag, about which the docs say:

> Doesn't append "api/v2/package" to the source URL.

Which implies that by default, it appends that suffix. So we need to remove it in the `NUGET_SERVER_URL` environment variable (which is set in the workflow in the parent repo).
I could instead use `--no-service-endpoint` and leave the URL unchanged, but I felt it was better to have just `https://www.nuget.org/` as the `NUGET_SERVER_URL`. I can change it if you prefer.

Comment (blairconrad) [reply FakeItEasy.Deploy/Program.cs:112]: I like your choice!

**Review by blairconrad — APPROVED** [review]
Thanks, @thomaslevesque. I assume you have a PR in the wings for FakeItEasy.Analyzers submodule and environment variable, so I'll stay out of your way.

**Comment by thomaslevesque** [comment]
> I assume you have a PR in the wings for FakeItEasy.Analyzers submodule and environment variable, so I'll stay out of your way.

I do. You read my mind!

**Pull request merged**